Implement uri parsing for DSMO

relud opened the pull request:
fixes #611

jklukas commented on ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseUri.java:
```diff
@@ -54,13 +72,166 @@ private ParseUri() {
+      if (unexpectedElements != 0) {
+        throw new UnexpectedPathElementsException(unexpectedElements);
+      }
+      // Initialize new payload with ping version and funnelcake
```
Is there documentation we can link here for what funnelcake is?

relud replied:
hmm, seems like https://wiki.mozilla.org/Funnelcake should do
```suggestion
      // Initialize new payload with ping version and funnelcake ID
      // https://wiki.mozilla.org/Funnelcake
```

relud commented:
> I'd love to inject more context here so that somebody looking at this code for the first time can easily understand what a "stub" is, what "DSMO" is/was, what a funnelcake is 🍰, and hopefully see a reference of the old logic that this is reimplementing.

Added a doc string to the stub uri class, and moved it to the end of ParseUri so that it interrupts the rest of the logic less.

> Is there any plan to move off of this logic and have stub installer send a normal structured ingestion payload, or will this need to be maintained for the long-term?

302 -> @matt-howell

matt-howell commented:
> Is there any plan to move off of this logic and have stub installer send a normal structured ingestion payload, or will this need to be maintained for the long-term?

There's no current plan. I don't have any objection to making that change, the problem is that old stub installers tend to stay in use basically indefinitely, so we wouldn't really be able to fully switch over for a long time without dropping some number of pings.
